mne/time_frequency/psd.py

```diff
@@ -51,22 +52,23 @@ def compute_raw_psd(raw, tmin=0., tmax=None, picks=None, fmin=0,
         The PSD for all channels
     freqs: array of float
         The frequencies
+
+    See Also
+    --------
+    psd_welch, psd_multitaper
     """
     from scipy.signal import welch
+    from ..io.base import _BaseRaw
+    if not isinstance(raw, _BaseRaw):
+        raise ValueError('Input must be an instance of Raw')
     tmax = raw.times[-1] if tmax is None else tmax
     start, stop = raw.time_as_index([tmin, tmax])
-    if picks is not None:
-        data, times = raw[picks, start:(stop + 1)]
-    else:
-        data, times = raw[:, start:(stop + 1)]
-    n_fft, n_overlap = _check_nfft(len(times), n_fft, n_overlap)
+    picks = slice(None) if picks is None else picks
 
     if proj:
-        proj, _ = make_projector_info(raw.info)
-        if picks is not None:
-            data = np.dot(proj[picks][:, picks], data)
-        else:
-            data = np.dot(proj, data)
+        raw = raw.apply_proj()
```

Comment:
It operates inplace so the assignment isn't necessary.

...and that's actually why it was done the old way before. You don't want to modify the input object like this. So please restore the old way of doing it. Except that we need to fix it -- we need to apply the projector, *then* subselect channels. The way it was done before -- by subselecting channels from the projector operator -- was wrong. This deserves a note in the `BUG` section of `whats_new.py`.